pkg/driver/driver_azure.go

```diff
@@ -167,14 +171,35 @@ func (d *AzureDriver) getVMParameters(vmName string, networkInterfaceReferenceID
 		VMParameters.Identity = &compute.VirtualMachineIdentity{
 			Type: compute.ResourceIdentityTypeUserAssigned,
 			UserAssignedIdentities: map[string]*compute.VirtualMachineIdentityUserAssignedIdentitiesValue{
-				*d.AzureMachineClass.Spec.Properties.IdentityID: &compute.VirtualMachineIdentityUserAssignedIdentitiesValue{},
+				*d.AzureMachineClass.Spec.Properties.IdentityID: {},
 			},
 		}
 	}
 
 	return VMParameters
 }
 
+func (d *AzureDriver) generateDataDisks(vmName string, azureDataDisks []v1alpha1.AzureDataDisk) []compute.DataDisk {
+	var dataDisks []compute.DataDisk
+	for _, azureDataDisk := range azureDataDisks {
+		dataDiskName := dependencyNameFromVMNameAndDependency(getAzureDataDiskName(azureDataDisk), vmName, dataDiskSuffix)
+		dataDiskLun := azureDataDisk.Lun
+		dataDiskSize := azureDataDisk.DiskSizeGB
+		dataDisk := compute.DataDisk{
+			Lun:     &dataDiskLun,
+			Name:    &dataDiskName,
+			Caching: compute.CachingTypes(azureDataDisk.Caching),
+			ManagedDisk: &compute.ManagedDiskParameters{
+				StorageAccountType: compute.StorageAccountTypes(azureDataDisk.ManagedDisk.StorageAccountType),
+			},
+			DiskSizeGB:   &dataDiskSize,
+			CreateOption: compute.DiskCreateOptionTypes(azureDataDisk.CreateOption),
```

Comment:
Do we need other CreateOptions than `Empty`? I think `Attach` and `FromImage` would require addional information like which existing volume should be attached or which image should be used to create the volume?

Reply:
I believe, this is also resolved, can you please confirm @guydaich ?

Reply:
I believe, this is also resolved, can you please confirm @guydaich @dkistner ?

Reply:
resolved

Reply:
confirmed